### PR TITLE
Add maintenance_tasks engine for running one-off tasks

### PR DESCRIPTION
**Pull request by segiddins**
Also, set up auditing so audits are created when tasks are manually kicked off via the admin web interface

This will make running backfills much easier, as maintainers will not have to babysit long-running rake tasks, and we will get proper error reporting

See https://github.com/Shopify/maintenance_tasks

### CODE REVIEW

**Review by indirect — APPROVED** [review]
Nice. Do we need to do any additional work to support ActiveJob? Are we already using GoodJob as our ActiveJob backend?

**Comment by simi** [comment]
> Are we already using GoodJob as our ActiveJob backend?

yes :partying_face: 

https://github.com/rubygems/rubygems.org/blob/9e93a64ffc1515179bd1ee8a9d4ff5aea9712f96/config/application.rb#L57

File: app/avo/resources/maintenance_tasks_run_resource.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class MaintenanceTasksRunResource < Avo::BaseResource
+  self.title = :id
+  self.includes = []
+  self.model_class = ::MaintenanceTasks::Run
+  # self.search_query = -> do
+  #   scope.ransack(id_eq: params[:q], m: "or").result(distinct: false)
+  # end
```
Comment (jenshenny) [inline app/avo/resources/maintenance_tasks_run_resource.rb:7]: What is the purpose of `search_query` here?

File: config/routes.rb [hunk 2]
```diff
@@ -270,6 +270,7 @@
     constraints(Constraints::Admin) do
       namespace :admin, constraints: Constraints::Admin::RubygemsOrgAdmin do
         mount GoodJob::Engine, at: 'good_job'
+        mount MaintenanceTasks::Engine, at: "maintenance_tasks"
```
Comment (jenshenny) [inline config/routes.rb:273]: nit: Can add a button on `http://localhost:3000/admin/resources/maintenance_tasks_runs` to the maintenance_tasks dashboard for easier access

**Review by jenshenny — APPROVED** [review]
Yay maintenance tasks! 🚀

File: app/avo/resources/maintenance_tasks_run_resource.rb (hunk 1 above)
Comment (segiddins) [reply app/avo/resources/maintenance_tasks_run_resource.rb:7]: it's commented out in the avo generator template, but it's for making resources searchable

File: config/routes.rb (hunk 2 above)
Comment (segiddins) [reply config/routes.rb:273]: There's no easy way to add any component to an avo index page as far as I can tell

**Pull request merged**